src/TestExplorer/TestRunner.ts

```diff
@@ -233,138 +184,129 @@ export class TestRunner {
+                outputStream.end();
+                return;
             }
-        } else {
-            const testBuildConfig = createTestConfiguration(this.folderContext, true);
+
+            // Watch the pipe for JSONL output and parse the events into test explorer updates.
+            // The await simply waits for the watching to be configured.
+            await this.swiftTestOutputParser.watch(fifoPipePath, runState);
+
+            await this.launchTests(
+                testKind,
+                token,
+                outputStream,
+                outputStream,
+                testBuildConfig,
+                runState
+            );
+
+            await asyncfs.rm(fifoPipePath, { force: true });
+        }
+
+        if (this.testArgs.hasXCTests) {
+            const testBuildConfig = LaunchConfigurations.createLaunchConfigurationForXCTestTesting(
+                this.testArgs.xcTestArgs,
+                this.workspaceContext,
+                this.folderContext,
+                false
+            );
             if (testBuildConfig === null) {
-                return null;
+                return;
             }
+            const testRegex = this.testRegex;
+            // Parse output from stream and output to log
+            const parsedOutputStream = new stream.Writable({
+                write: (chunk, encoding, next) => {
+                    const text = chunk.toString();
+                    this.testRun.appendOutput(text.replace(/\n/g, "\r\n"));
+                    this.xcTestOutputParser.parseResult(text, runState, testRegex);
+                    next();
+                },
+            });
 
-            if (testList.length > 0) {
-                testBuildConfig.args = [testList];
+            // Output test from stream
+            const outputStream = new stream.Writable({
+                write: (chunk, encoding, next) => {
+                    const text = chunk.toString();
+                    this.testRun.appendOutput(text.replace(/\n/g, "\r\n"));
+                    next();
+                },
+            });
+
+            if (token.isCancellationRequested) {
+                parsedOutputStream.end();
+                outputStream.end();
+                return;
             }
-            // output test logging to debug console so we can catch it with a tracker
-            testBuildConfig.terminal = "console";
-            return testBuildConfig;
+
+            await this.launchTests(
+                testKind,
+                token,
+                parsedOutputStream,
+                outputStream,
+                testBuildConfig,
+                runState
+            );
         }
     }
 
-    /** Run test session without attaching to a debugger */
-    async runSession(
-        token: vscode.CancellationToken,
+    private async launchTests(
         testKind: TestKind,
+        token: vscode.CancellationToken,
+        stdout: stream.Writable,
+        stderr: stream.Writable,
```

Comment:
Calling them stdout and stderr is a little confusing as the first one is the output stream that is parsed and the second one is the output stream that is just passed onto the test runner output. On macOS the parsed output stream is stderr, on Linux it is stdout. Well as least for XCTest